base/src/main/java/io/spine/query/QueryPredicate.java

```diff
@@ -427,18 +427,18 @@ private void optimizeForOnlyChild() {
          * the same logical operator as this builder.
          */
         private void flattenSimilarChildren() {
-                long differentChildrenCount =
-                        children.stream()
-                                .filter(c -> c.operator != operator())
-                                .count();
-                if(differentChildrenCount == 0) {
-                    for (QueryPredicate<R> child : children) {
-                        copyParams(child, this);
-                        copyCustomParams(child, this);
-                        copyChildren(child, this);
-                    }
-                    children.clear();
+            long differentChildrenCount =
+                    children.stream()
+                            .filter(c -> c.operator != operator())
```

Comment:
Do we need to call the `operator()` method for each child? Can we extract the result of the call into a variable instead?

Reply:
@alexander-yevsyukov 
This is actually a plain getter returning a constant value. So `javac` is going to replace it with the value anyway. 
I was thinking of shortening the method rather than of explicit optimisations, such as extracting into a variable.

@dmdashenkov thanks for formatting this piece!